packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { screen, render, waitFor, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ApiTokenContext from '../api-token-context';
+import ApiTokenDeleteButton from '../api-token-delete-button';
+
+const modal_root_el = document.createElement('div');
+modal_root_el.setAttribute('id', 'modal_root');
+document.body.appendChild(modal_root_el);
+
+describe('ApiTokenDeleteButton', () => {
+    const mock_props = {
+        api_tokens: [
+            {
+                display_name: '',
+                last_used: '',
+                scopes: [],
+                token: '',
+            },
+        ],
+        deleteToken: jest.fn(() => Promise.resolve()),
+        footer_ref: document.createElement('div'),
+        overlay_ref: document.createElement('div'),
+        toggleOverlay: jest.fn(),
+    };
+    const mock_token = {
+        token: {
+            display_name: 'Token 1',
+            last_used: '12/31/2022',
+            scopes: ['read', 'trade'],
+            token: '1234567',
+        },
+    };
+
+    const renderAPIDeleteButton = () => {
+        render(
+            <ApiTokenContext.Provider value={mock_props}>
+                <ApiTokenDeleteButton {...mock_token} />
+            </ApiTokenContext.Provider>
+        );
+    };
+
+    it('should render ApiTokenDeleteButton', () => {
+        renderAPIDeleteButton();
+        expect(screen.getByTestId('dt_token_delete_icon')).toBeInTheDocument();
+        expect(screen.queryByText('Delete this token')).not.toBeInTheDocument();
+    });
+
+    it('should display Delete this token when mouse enter', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+    });
+
+    it('should not Delete this token when mouse leave', () => {
```

Comment:
As per test case, we are checking 'should not display Delete this token message when mouse leave'. can you check the test case here